src/main/java/com/ibm/devops/connect/CloudPublisher.java

```diff
@@ -314,103 +353,106 @@ public static boolean uploadQualityData(HttpEntity entity, String url, String us
             }
         }
     }
 
-    public static void uploadQualityDataRaw(String payload) throws Exception {
+    public static void uploadQualityDataRaw(int instanceNum,String payload) throws Exception {
         CloudPublisher.ensureHttpClientInitialized();
-        String localLogPrefix= logPrefix + "uploadMetricDataRaw ";
         String resStr = "";
-        String url = CloudPublisher.getQualityDataRawUrl();
+        String url = CloudPublisher.getQualityDataRawUrl(instanceNum);
         CloseableHttpResponse response = null;
+        List<Entry> entries = Jenkins.getInstance().getDescriptorByType(DevOpsGlobalConfiguration.class).getEntries();
 
         try {
             HttpPost postMethod = new HttpPost(url);
-            attachHeaders(postMethod);
+            attachHeaders(postMethod, instanceNum);
             postMethod.setHeader("Content-Type", "application/json");
-            postMethod.setHeader("Authorization", "UserAccessKey " + Jenkins.getInstance().getDescriptorByType(DevOpsGlobalConfiguration.class).getApiToken());
+            postMethod.setHeader("Authorization", "UserAccessKey " + entries.get(instanceNum).getApiToken());
             postMethod.setEntity(new StringEntity(payload));
 
             response = httpClient.execute(postMethod);
             resStr = EntityUtils.toString(response.getEntity());
             if (response.getStatusLine().toString().contains("200")) {
-                log.info(localLogPrefix + "Uploaded Metric (raw) successfully");
+                log.info("[UrbanCode Velocity "+ entries.get(instanceNum).getBaseUrl() + "] CloudPublisher#uploadMetricDataRaw Uploaded Metric (raw) successfully");
             } else {
-                throw new Exception("Bad response code when uploading Metric (raw): " + response.getStatusLine() + " - " + resStr);
+                throw new Exception("[UrbanCode Velocity "+ entries.get(instanceNum).getBaseUrl() + "] CloudPublisher#uploadMetricDataRaw Bad response code when uploading Metric (raw): " + response.getStatusLine() + " - " + resStr);
             }
         } finally {
             if (response != null) {
                 try {
                     response.close();
                 } catch (Exception e) {
-                    log.error("Could not close uploadQualityDataRaw response");
+                    log.error("[UrbanCode Velocity "+ entries.get(instanceNum).getBaseUrl() + "] CloudPublisher#uploadMetricDataRaw Could not close uploadQualityDataRaw response");
                 }
             }
         }
     }
 
-    private static void attachHeaders(AbstractHttpMessage message) {
-        String syncId = Jenkins.getInstance().getDescriptorByType(DevOpsGlobalConfiguration.class).getSyncId();
-        message.setHeader("sync_token", Jenkins.getInstance().getDescriptorByType(DevOpsGlobalConfiguration.class).getSyncToken());
+    private static void attachHeaders(AbstractHttpMessage message, int instanceNum) {
```

Comment:
this could just take in the syncId and the syncToken as a param and not need to go through the entries each time